query/control/controller_test.go

```diff
@@ -122,6 +122,15 @@ func validateUnusedMemory(t testing.TB, reg *prometheus.Registry, c control.Conf
 	}
 }
 
+func TestConfig_ValidateMaxConcurrencyQuota(t *testing.T) {
+	config.ConcurrencyQuota = control.MaxQueryConcurrency + 1
```

Comment:
My Golang ignorance - does this correctly cast MazQueryConcurrency to an int64 before the addition, and void a wrap-around?

Reply:
Once I changed the type of `ConcurrencyQuota` to `int32`:
1. This still compiled fine
2. Running the test resulted in immediate failure, with an overflow error